cachi2/core/package_managers/yarn_classic/resolver.py

```diff
@@ -47,26 +53,99 @@ class _RelpathMixin(BaseModel):
 class RegistryPackage(_BasePackage, _UrlMixin):
     """A Yarn 1.x package from the registry."""
 
+    @property
+    def purl(self) -> str:
+        """Return package URL."""
+        qualifiers = {}
+
+        if self.url != DEFAULT_NPM_REGISTRY:
```

Comment:
We do have a check for that:
https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/package_managers/yarn_classic/resolver.py#L184

IOW, if a package is `RegistryPackage`  it is one of those npm registries.

Reply:
Thank you for pointing me to this. Could you please add this explicitly? Essentially like this:
```python
class RegistryPackage(_BasePackage, _UrlMixin):
    """A Yarn 1.x package from the registry."""
    url: Literal[*NPM_REGISTRY_CNAMES]
```
It will significantly reduce the number of such questions in the future.

Reply:
Should the URL be like `"https://registry.yarnpkg.com/foo/-/foo-1.0.0.tgz#fffffff"` or just `https://registry.yarnpkg.com` ? :thinking: cc @taylormadore

Reply:
Currently, we store `https://registry.yarnpkg.com/foo/-/foo-1.0.0.tgz#fffffff` as the `url` attribute so I would need to change the implementation slightly.